python/lsst/scarlet/lite/models/free_form.py

```diff
@@ -126,10 +130,119 @@ def resize(self, model_box: Box) -> bool:
 
     def __str__(self):
         return (
-            f"FreeFormComponent(\n    bands={self.bands}\n    "
+            f"FactorizedFreeFormComponent(\n    bands={self.bands}\n    "
             f"spectrum={self.spectrum})\n    center={self.peak}\n    "
             f"morph_shape={self.morph.shape}"
         )
 
     def __repr__(self):
         return self.__str__()
+
+
+class FreeFormComponent(Component):
+    """Implements a free-form component
+
+    This is a FreeFormComponent that is not factorized into a
+    spectrum and morphology with no monotonicity constraint.
+    """
+
+    def __init__(
+        self,
+        bands: tuple,
+        model: np.ndarray | Parameter,
+        model_bbox: Box,
+        bg_thresh: float | None = None,
+        bg_rms: np.ndarray | None = None,
+        floor: float = 1e-20,
+        peaks: list[tuple[int, int]] | None = None,
+        min_area: float = 0,
+    ):
+        if len(bands) != 1:
+            raise ValueError("MonochromaticDeconvolvedComponent only supports one band")
+        super().__init__(bands=bands, bbox=model_bbox)
+        self._model = parameter(model)
+        self.bg_rms = bg_rms
+        self.bg_thresh = bg_thresh
+        self.floor = floor
+        self.peaks = peaks
+        self.min_area = min_area
+
+    @property
+    def model(self) -> np.ndarray:
+        """The morphological model of the component"""
+        return self._model.x
+
+    def get_model(self) -> Image:
+        """Convert the model into an image"""
+        return Image(self.model, bands=self.bands, yx0=cast(tuple[int, int], self.bbox.origin))
+
+    @property
+    def shape(self) -> tuple:
+        """Shape of the resulting model image"""
+        return self.model.shape
+
+    def grad_model(self, input_grad: np.ndarray, model: np.ndarray) -> np.ndarray:
+        """Gradient of the morph wrt. the component model"""
+        return input_grad
+
+    def prox_model(self, model: np.ndarray) -> np.ndarray:
+        """Apply a prox-like update to the model"""
+        from lsst.scarlet.lite.detect_pybind11 import get_connected_multipeak, get_footprints  # type: ignore
```

Comment:
Still confused by this pattern, but I suppose you can answer the first time I asked.

Reply:
Yeah, there was definitely an issue with loading this at the top of the file, either a circular dependency or a mypy issue, or something? I'm still baffled by why it used to fail before but works now, but regardless, it works now at the beginning of the file so I moved it.